### ISSUE TITLE
@typescript-eslint/no-useless-constructor won't turn off.

### ISSUE THREAD

**Issue by gpresland** [typescript-eslint/typescript-eslint#873]
**What code were you trying to parse?**

```ts
  constructor() {
    super();
  }
```

with _.eslintrc.js_

```
  rules: {
    "no-useless-constructor": "off",
    "@typescript-eslint/no-useless-constructor": false
  }
```

and _tslint.json_

```
  "jsRules": {
    "no-useless-constructor": "off",
    "@typescript-eslint/no-useless-constructor": false
  },
  "rules": {
    "no-useless-constructor": "off",
    "@typescript-eslint/no-useless-constructor": false
  },
```

along with every value combinations off `"off"`, `false`, and `0`.

Documentation followed at:

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-useless-constructor.md

**What did you expect to happen?**

Expect to receive no warnings for _useless constructors_.

**What actually happened?**

Compiled with warnings.

```
Line 31:  Useless constructor  @typescript-eslint/no-useless-constructor
```

**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin-tslint` | `1.6.0` |
| `@typescript-eslint/parser`               | `1.6.0` |
| `TypeScript`                              | `3.5.2` |
| `ESLint`                                  | `5.16.0` |
| `node`                                    | `10.16.0` |
| `npm`                                     | `6.9.0` |

**Comment by bradzacher** [comment]
Could you post your eslint config as well?

Chances are you're using a config set which uses `overrides` to apply the rule to typescript files only. This has been fixed in eslint6.

I also notice you're using v1.6 of our tooling, why are you on such an old version? You should upgrade as you're missing a large number of bug fixes and performance improvements.

**Comment by gpresland** [comment]
@bradzacher 

This was all created roughly a month ago with create-react-app and Storybook. The version of 1.6.0 in the tooling is what I see for the _version_ value in the _package.json_ under _node_modules\@typescript-eslint_ that create-react-app installed. I didn't install these modules manually, create-react-app installs them automatically.

A fresh test of create-react-app appears to install version 1.13.0.

Looking through the files after your input, I am not sure why I have an _.eslintrc_ file as all files are TypeScript and am using TSLint. I've deleted my _.eslintrc_ file and the problem continues to persists.

Here is the _.eslintrc_ anyways though as requested:
 
```
module.exports = {
  env: {
    browser: true,
    es6: true,
  },
  extends: [
    "airbnb",
  ],
  globals: {
    Atomics: "readonly",
    SharedArrayBuffer: "readonly",
  },
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 2018,
    sourceType: "module",
  },
  plugins: [
    "react",
  ],
  rules: {
    "comma-dangle": 0,
    "global-require": 0,
    "import/no-unresolved": 0,
    "react/jsx-filename-extension": 0,
    "no-useless-constructor": 0,
    "@typescript-eslint/no-useless-constructor": 0
  }
};
```

**Comment by bradzacher** [comment]
`create-react-app` provides config by default.
Even if you provide it non, or you provide it your own, it still provides its own.

This is the config: https://github.com/facebook/create-react-app/blob/master/packages/eslint-config-react-app/index.js

Note that, as expected, it uses an `overrides` block to apply the rule to typescript files only.

You will have to use a similar overrides block in your config because you're using eslint v5.
